mesonbuild/backend/ninjabackend.py

```diff
@@ -823,7 +823,7 @@ def generate_tests(self):
             cmd += ['--no-stdsplit']
         if self.environment.coredata.get_builtin_option('errorlogs'):
             cmd += ['--print-errorlogs']
-        elem = NinjaBuildElement(self.all_outputs, 'meson-test', 'CUSTOM_COMMAND', ['all', 'PHONY'])
+        elem = NinjaBuildElement(self.all_outputs, 'meson-test', 'CUSTOM_COMMAND', ['all', 'tests', 'PHONY'])
```

Comment:
I don't think this should depend on `all`, that's likely to hide bugs in your meson.build or trigger unnecessary rebuilds when testing.

Reply:
Well possible - I never thought about trying to drop the 'all' dependency.